mismatched types - enif_make_long(env, i) ^ expected i32, found i64

Hey guys! In my excitement at the lovely improvements in #254 I'm trying to build against the current master branch (revision 5da90198e) 

I'm getting a weird error in cargo build, `error: Could not compile 'rustler_sys'.`  

There are 4 errors about mismatched types in rustler_sys nif_api.snippet 

**Do you have any suggestions how I could fix this please?** 

```
error[E0308]: mismatched types
   --> C:\path\to\my_app\native\mynif\target\debug\build\rustler_sys-7ebc48992530b824\out/nif_api.snippet:907:27
    |
907 |     { enif_make_long(env, i) }
    |                           ^ expected i32, found i64
help: you can convert an `i64` to `i32` and panic if the converted value wouldn't fit
    |
907 |     { enif_make_long(env, i.try_into().unwrap()) }
    |                           ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> C:\path\to\my_app\native\mynif\target\debug\build\rustler_sys-7ebc48992530b824\out/nif_api.snippet:912:28
    |
912 |     { enif_make_ulong(env, i) }
    |                            ^ expected u32, found u64
help: you can convert an `u64` to `u32` and panic if the converted value wouldn't fit
    |
912 |     { enif_make_ulong(env, i.try_into().unwrap()) }
    |                            ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> C:\path\to\my_app\native\mynif\target\debug\build\rustler_sys-7ebc48992530b824\out/nif_api.snippet:917:32
    |
917 |     { enif_get_long(env, term, ip) }
    |                                ^^ expected i32, found i64
    |
    = note: expected type `*mut i32`
               found type `*mut i64`

error[E0308]: mismatched types
   --> C:\path\to\my_app\native\mynif\target\debug\build\rustler_sys-7ebc48992530b824\out/nif_api.snippet:922:33
    |
922 |     { enif_get_ulong(env, term, ip) }
    |                                 ^^ expected u32, found u64
    |
    = note: expected type `*mut u32`
               found type `*mut u64`

error: aborting due to 4 previous errors
```

I'm on windows but hopefully that's not an issue - rustler 0.21 works perfectly 

Thank you for your time and this cool project!

--- a/rustler_sys/build.rs
+++ b/rustler_sys/build.rs
@@ -27,12 +27,13 @@ fn try_gen_api(dst: &Path, pointer_size: usize) -> bool {
 }
 
 fn main() {
-    // get size of C pointer
-    let target_pointer_width = match env::var("CARGO_CFG_TARGET_POINTER_WIDTH") {
-        Ok(ref val) if val == "32" => 4,
-        Ok(ref val) if val == "64" => 8,
-        Ok(ref val) => panic!("Unsupported target pointer width: {}", val),
-        Err(err) => panic!(
+    // get size of C long
+    let long_width = match (env::var("CARGO_CFG_TARGET_POINTER_WIDTH"), env::var("CARGO_CFG_TARGET_OS")) {
+        (_, Ok(ref os)) if os == "windows" => 4,
+        (Ok(ref val), _) if val == "32" => 4,
+        (Ok(ref val), _) if val == "64" => 8,
+        (Ok(ref val), _) => panic!("Unsupported target pointer width: {}", val),
+        (Err(err), _) => panic!(
             "An error occurred while determining the pointer width to compile `rustler_sys` for:\n\n{:?}\n\nPlease report a bug.",
             err
         ),
@@ -45,11 +46,11 @@ fn main() {
 
     let dst = Path::new(&out_dir).join(SNIPPET_NAME);
 
-    if !try_gen_api(&dst, target_pointer_width) {
+    if !try_gen_api(&dst, long_width) {
         eprintln!("Failed to generate API from local installation, falling back to precompiled");
 
         let source =
-            Path::new(&"precompiled").join(format!("nif_api.{}.snippet", target_pointer_width));
+            Path::new(&"precompiled").join(format!("nif_api.{}.snippet", long_width));
 
         fs::copy(&source, &dst).unwrap();
     }

--- a/rustler_sys/build.rs
+++ b/rustler_sys/build.rs
@@ -28,7 +28,10 @@ fn try_gen_api(dst: &Path, pointer_size: usize) -> bool {
 
 fn main() {
     // get size of C long
-    let long_width = match (env::var("CARGO_CFG_TARGET_POINTER_WIDTH"), env::var("CARGO_CFG_TARGET_OS")) {
+    let target_pointer_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH");
+    let target_os = env::var("CARGO_CFG_TARGET_OS");
+
+    let long_width = match (target_pointer_width, target_os) {
         (_, Ok(ref os)) if os == "windows" => 4,
         (Ok(ref val), _) if val == "32" => 4,
         (Ok(ref val), _) if val == "64" => 8,
@@ -49,8 +52,7 @@ fn main() {
     if !try_gen_api(&dst, long_width) {
         eprintln!("Failed to generate API from local installation, falling back to precompiled");
 
-        let source =
-            Path::new(&"precompiled").join(format!("nif_api.{}.snippet", long_width));
+        let source = Path::new(&"precompiled").join(format!("nif_api.{}.snippet", long_width));
 
         fs::copy(&source, &dst).unwrap();
     }

